tests/test_data.py

```diff
@@ -28,18 +31,23 @@ def test_sqldataset_repr_prints_correctly(test_sqldata):
     assert repr == "<BostonData - SQLDataset Engine(sqlite:///:memory:)>"
 
 
-def test_sqldataset_can_be_instantiated_with_engine_string():
-    class BostonDataSet(SQLDataset):
-        def load_training_data(self, *args, **kwargs):
-            pass
+def test_sqldataset_can_be_instantiated_with_engine_string(test_sqldata_class):
+    assert test_sqldata_class("sqlite:///", "")
 
-        def load_prediction_data(self, *args, **kwargs):
-            pass
+    with pytest.raises(ValueError, match="Invalid connection"):
+        test_sqldata_class(["not_a_url"], "schema")
 
-    assert BostonDataSet("sqlite:///")
 
-    with pytest.raises(ValueError, match="Invalid connection"):
-        BostonDataSet(["not_a_url"])
+def test_sqldataset_errors_when_schema_is_defined_on_instantiation(
+    test_sqldata_class, test_db
+):
+    test_sqldata_class.table = MagicMock()
+
+    with pytest.raises(
+        DatasetError,
+        match="cannot have a defined schema - remove the schema declaration",
+    ):
+        assert test_sqldata_class(test_db, "schema")
```

Comment:
```suggestion
    test_sqldata_class(test_db, "schema")
```
Don't need an assert in a `pytest.raises` block